modules/dxp/apps/search-experiences/search-experiences-web/src/test/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommandTest.java

```diff
@@ -0,0 +1,397 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import javax.portlet.PortletURL;
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+/**
+ * @author Felipe Lorenz
+ */
+public class GetSitesMVCResourceCommandTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() throws Exception {
+		_getSitesMVCResourceCommand = new GetSitesMVCResourceCommand();
+
+		_setUpThemeDisplay(_resourceRequest);
+
+		ReflectionTestUtil.setFieldValue(
+			_getSitesMVCResourceCommand, "_groupService", _groupService);
+
+		_paramUtilMockedStatic = Mockito.mockStatic(ParamUtil.class);
+	}
+
+	@After
+	public void tearDown() {
+		_paramUtilMockedStatic.close();
+	}
+
+	@Test
+	public void testGetSiteByExternalReferenceCodeJSONObject()
+		throws Exception {
+
+		Group childrenGroup = _createGroup(
+			RandomTestUtil.randomString(), true, new ArrayList<>());
+
+		_setUpParamUtil("externalReferenceCode", RandomTestUtil.randomString());
+
+		Mockito.doReturn(
+			childrenGroup
+		).when(
+			_groupService
+		).fetchGroupByExternalReferenceCode(
+			Mockito.anyString(), Mockito.anyLong()
+		);
+
+		_setUpGroups(
+			true, 3,
+			new ArrayList<Group>() {
+				{
+					add(childrenGroup);
+				}
+			});
+
+		JSONObject siteJSONObject =
+			_getSitesMVCResourceCommand.
+				getSiteByExternalReferenceCodeJSONObject(_resourceRequest);
+
+		Assert.assertEquals(
+			siteJSONObject.toString(),
+			childrenGroup.getDescriptiveName(_themeDisplay.getLocale()),
+			siteJSONObject.get("descriptiveName"));
+		Assert.assertEquals(
+			siteJSONObject.toString(), childrenGroup.getExternalReferenceCode(),
+			siteJSONObject.get("externalReferenceCode"));
+		Assert.assertEquals(
+			siteJSONObject.toString(),
+			String.valueOf(childrenGroup.getGroupId()),
+			siteJSONObject.get("groupId"));
+		Assert.assertEquals(
+			siteJSONObject.toString(),
+			childrenGroup.getName(_themeDisplay.getLocale()),
+			siteJSONObject.get("name"));
+	}
+
+	@Test
+	public void testGetSitesJSONObjectWithChildrenGroups() throws Exception {
+		_setUpPages(_resourceRequest, 10, 1);
+
+		Group childrenGroup = _createGroup(
+			RandomTestUtil.randomString(), true, new ArrayList<>());
+
+		_setUpGroups(
+			true, 1,
+			new ArrayList<Group>() {
+				{
+					add(childrenGroup);
+				}
+			});
+
+		JSONObject sitesJSONObject =
+			_getSitesMVCResourceCommand.getSitesJSONObject(_resourceRequest);
+
+		JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("items");
+
+		Assert.assertEquals(
+			sitesJSONArray.toString(), 2, sitesJSONArray.length());
+	}
+
+	@Test
+	public void testGetSitesJSONObjectWithDifferentCompanyGroupId()
```

Comment:
`testGetSitesJSONObjectWithDifferentCompanyGroupId` should be renamed `testGetSitesJSONObjectFromNonActiveGroups`